### PR TITLE
[TFlite][CPU] Import TFLite custom nodes and generate external function calls

### PR DESCRIPTION
**Pull request by mciprian13**
**Summary**
- Add support in InstrGen for variadic operands (at constructor time):
  - At this point automatic IRGen is not support for these kind of instructions (manually implemented). 
- Add support in TFLiteModelLoader to import custom operators by creating a `TFLiteCustomOperator` node
- Create in CPU backend an external call to a user-defined callback:
  - The options of the custom operator are provided to the callback as-is (unformatted raw format). The parsing of the options (most commonly in flexbuffer format) is done in the callback.

**Test Plan**
Add a unit test in the EXPENSIVE test suite which compiles a TFLite model with a custom operator into a bundle using the `model-compiler` tool and links with an application which implements the callback and prints the information.

You can find some more details about the associated NodeGen/InstrGen and unit testing in this archive:
[PR_tflite_custom_node_code_gen.zip](https://github.com/pytorch/glow/files/9039520/PR_tflite_custom_node_code_gen.zip)

### CODE REVIEW

**Comment by mciprian13** [comment]
@opti-mix Can you take a look on this PR? I think you want to take a look because this PR introduces variadic operands for Intructions and also tests the previously introduced mechanism for variadic operands for Nodes.

**Review by opti-mix — COMMENTED** [review]
@mciprian13 Nice feature! Thanks! Generally it looks pretty good.

I'll try to import it into our internal build system and see if it breaks anything.

File: tools/ClassGen/InstrBuilder.cpp [hunk 1]
```diff
@@ -35,29 +35,57 @@ void InstrBuilder::emitCtor(std::ostream &os) const {
     os << ", Value *" << op.first;
   }
 
+  // The variable operands of the instruction class:
+  for (const auto &op : variableOperands_) {
+    os << ", std::vector<Value *> " << op.first;
```
[the remark this answers is not in the document]
Comment (mciprian13) [reply tools/ClassGen/InstrBuilder.cpp]: I changed it to `llvm::ArrayRef`. Done.

File: lib/IR/IRGen.cpp [hunk 2]
```diff
@@ -110,6 +110,45 @@ void IRGenVisitor::post(Node *parent, Node *N) {
     // Include all automatically generated cases:
 #include "glow/AutoGenIRGen.h"
 
+  case glow::Kinded::Kind::TFLiteCustomOperatorNodeKind: {
+    auto *TCON = cast<TFLiteCustomOperatorNode>(N);
+
+    // Get instruction inputs.
+    size_t numInputs = TCON->getNumInputs();
+    std::vector<Value *> inputs(numInputs);
```
[the remark this answers is not in the document]
Comment (mciprian13) [reply lib/IR/IRGen.cpp]: I changed both inputs and outputs to `llvm::SmallVector<Value *, 8>`. Done.

**Comment by mciprian13** [comment]
@opti-mix Are there problems I need to fix?

**Comment by mciprian13** [comment]
@opti-mix Thanks for reviewing and landing this!

**Pull request closed without merging**